### PR TITLE
chore: update nuget build and deploy github action to support semantic versioning 2.0

### LINKED ISSUE #578: [chore]: update the GitHub nuget build and deploy action to work with Semantic Version 2
**Is your feature request related to a problem? Please describe.**

Currently the [Build and Deploy Nuget package](https://github.com/appium/dotnet-client/blob/master/.github/workflows/release-nuget.yml) GitHub action is failing and does not pick up beta releases.

**Describe the solution you'd like**
- [ ] A GitHub Action builds the solution and deploys to the nuget.org repository
- [ ] An updated matcher to match published release (versions/tags including pre-releases according to Semantic Version 2.0)

### PR DESCRIPTION
**Pull request by laolubenson**
## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New feature (non-breaking change which adds value to the project)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
Updates the [GitHub nuget build and deploy action](https://github.com/appium/dotnet-client/actions/workflows/release-nuget.yml) to work with Semantic Version 2
resolves #578

### CODE REVIEW

**Comment by Dor-bl** [comment]
@akinsolb seems like you have a merge conflict.

**Comment by akinsolb** [comment]
> @akinsolb seems like you have a merge conflict.

Ah yes, this may be because another pr got merged before this. I'll fix in a moment

**Review by Dor-bl — CHANGES_REQUESTED** [review]
What about the net45 removal?

File: test/integration/Appium.Net.Integration.Tests.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net48;net45</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (laolubenson) [reply test/integration/Appium.Net.Integration.Tests.csproj:3]: This was also removed to fix the build on github actions. It doesn't have the runtime pack, nor is it attainable.

**Review by Dor-bl — APPROVED** [review]

**Pull request merged**